### ISSUE TITLE
[.NET 8.0 Headless] websocket-sharp does not support .NET 8

### ISSUE THREAD

**Issue by WallTack** [Yellow-Dog-Man/Resonite-Issues#2586]
### Describe the issue.

Attempting to connect a WebsocketClient while the HandlingUser is set to a .NET 8 headless client fails to connect and throws the following error:

```
Exception running asynchronous task:
System.AggregateException: One or more errors occurred. (Operation is not supported on this platform.)
 ---> System.PlatformNotSupportedException: Operation is not supported on this platform.
   at System.Func`1.BeginInvoke(AsyncCallback callback, Object object)
   at WebSocketSharp.WebSocket.ConnectAsync()
   at FrooxEngine.WebsocketClient.ConnectTo(Uri target)
   at FrooxEngine.WebsocketClient.<OnChanges>b__25_0()
   --- End of inner exception stack trace ---

   at System.Environment.get_StackTrace()
   at Elements.Core.UniLog.Error(String message, Boolean stackTrace)
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()
```

It seems like the long term solution is to migrate away from the dependency on websocket-sharp, but [this workaround](https://github.com/sta/websocket-sharp/issues/652#issuecomment-1092129250) might be worth considering.

### To Reproduce

1. Start a .NET 8 headless using the default config but with `allowedUrlHosts` set to `["wss://echo.websocket.org"]`
2. Join the session
3. Create an empty object
4. Add a WebsocketClient component to the empty object
5. Set the HandlingUser of the WebsocketClient to the .NET 8 headless user
6. Set the URL of the WebsocketClient to `wss://echo.websocket.org/`
7. Observe the `Operation is not supported on this platform` error in the headless logs

### Expected behavior

The WebsocketClient should be able to connect and send/receive messages as it does when running under .NET Framework.

### Resonite Version Number

2024.7.15.1359

### What Platforms does this occur on?

Windows, Linux

### Log Files

Windows: [WALLTACK - 2024.7.15.1359 - 2024-07-15 16_00_41.log](https://github.com/user-attachments/files/16241161/WALLTACK.-.2024.7.15.1359.-.2024-07-15.16_00_41.log)
Linux: [WallTackHeadless - 2024.7.15.1359 - 2024-07-15 23_13_28.log](https://github.com/user-attachments/files/16241204/WallTackHeadless.-.2024.7.15.1359.-.2024-07-15.23_13_28.log)

### Additional Context

Both of the log files above were produced by following the steps to reproduce and then shutting down the headless using the `shutdown` command. Shutting down the headless at this point also produces another WebSocketSharp exception which can be seen at the end of both log files.

### Reporters

WallTack

**Comment by Frooxius** [comment]
I've tried a few alternative forks, but none of them seem to work. I'll have to dig into this a bit more and see what's going on.

**Comment by hazre** [comment]
> but [this workaround](https://github.com/sta/websocket-sharp/issues/652#issuecomment-1092129250) might be worth considering.

@WallTack I tried this by harmony patching the WebsocketClient Class in FrooxEngine.dll and replacing ConnectAsync with Connect and CloseAsync with Close. It doesn't work. Same error happens:

```
Exception running asynchronous task:
System.AggregateException: One or more errors occurred. (Operation is not supported on this platform.)
 ---> System.PlatformNotSupportedException: Operation is not supported on this platform.
   at System.Action`1.BeginInvoke(T obj, AsyncCallback callback, Object object)
   at WebSocketSharp.WebSocket.open()
   at WebSocketSharp.WebSocket.Connect()
   at websocketdotnet8fix.websocketdotnet8fix.WebsocketClientConnectToPatch.ConnectToReplacement(WebsocketClient instance, Uri target)
   at FrooxEngine.WebsocketClient.<OnChanges>b__25_0()
   --- End of inner exception stack trace ---

   at System.Environment.get_StackTrace()
   at Elements.Core.UniLog.Error(String message, Boolean stackTrace)
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(Thread threadPoolThread, ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading.PortableThreadPool.WorkerThread.WorkerThreadStart()
   ```

**Comment by Frooxius** [comment]
This should be fixed in 2024.7.22.1319, thanks for report!
